### PR TITLE
3d dataviz of gas prices vs confirmation time

### PR DESCRIPTION
**Pull request by owocki**
[see the output here](http://bits.owocki.com/0q2q452E2C0Z/out.mp4)

could be cool to put on the gas station, not sure how its gonna perform marketing-wise tho

### CODE REVIEW

**Comment by SaptakS** [comment]
@owocki this looks cool. So if I understand correctly, this rotates on mousedrag in 3d space??

PS: how about also giving interactivity as in how past back in the history the users are

**Comment by SaptakS** [comment]
Sorry. My mistake.

File: app/gas/management/commands/output_gas_viz.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+
+from gas.models import GasProfile
+from numpy import array
+
+
+def convert_to_movie():
+    command = "ffmpeg -framerate 30 -pattern_type glob -i 'cache/frames/*.jpg' -c:v libx264 -pix_fmt yuv420p cache/out.mp4"
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply app/gas/management/commands/output_gas_viz.py]: Thoughts on using an ffmpeg wrapper for this like `moviepy` or something similar?

File: app/gas/management/commands/output_gas_viz.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    print("converting to movie")
+    os.system(command)
+
+
+def install_ffmpeg():
+    command = 'apt update -y '
+    os.system(command)
+    command = 'apt install ffmpeg libav-tools x264 x265 -y'
```
Comment (mbeacom) [inline app/gas/management/commands/output_gas_viz.py]: Can we add `ffmpeg libav-tools x264 x265` to the dockerfile build steps?

File: app/gas/management/commands/output_gas_viz.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+            if vantage_point == 'low':
+                z_angle = 2
+            if vantage_point == 'med':
+                z_angle = 5
+            if vantage_point == 'high':
+                z_angle = 10
+            ax.view_init(z_angle, degrees)
+
+            png_file = 'cache/frames/{}.jpg'.format(str(j).rjust(10, '0'))
```
Comment (mbeacom) [inline app/gas/management/commands/output_gas_viz.py]: Can we update this to use an f-string?

File: app/retail/templates/gas.html [hunk 4]
```diff
@@ -52,119 +52,135 @@
...
+  </li>
+  <li>
+    <a href="{% url 'gas' %}?is_3d=1">View Historical in 3d</a>
+  </li>
+</ul>
+{% if 1 %}
+  <br>
+  <video width="640" height="480" controls>
+  <source src="/static/tmp/gas_price_viz.mp4" type="video/mp4">
+Your browser does not support the video tag.
```
Comment (mbeacom) [inline app/retail/templates/gas.html]: Should we wrap this in `{% trans "" %}` ?

File: app/retail/templates/gas.html [hunk 5]
```diff
@@ -52,119 +52,135 @@
...
+    svg.selectAll('circle').data(data)
+      .enter().append('circle')
+      .attr('class', 'dot')
+      .attr('r', 3)
+      .attr('cx', d => x(d.confirmation_time))
+      .attr('cy', d => y(d.gas_price))
+      .on('mouseover', function(d) {
+        var tooltip_msg = '<div>';
+        tooltip_msg = `Gas Price: ${d.gas_price}`;
+        tooltip_msg = tooltip_msg + `<br/>Confirmation time: ${d.confirmation_time} min`;
```
Comment (mbeacom) [inline app/retail/templates/gas.html]: Should we wrap the copy throughout in `trans` ?

File: app/gas/management/commands/output_gas_viz.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (owocki) [reply app/gas/management/commands/output_gas_viz.py]: either way seems acceptable to me.

File: app/gas/management/commands/output_gas_viz.py (hunk 3 above)
Comment (owocki) [reply app/gas/management/commands/output_gas_viz.py]: updated

File: app/gas/management/commands/output_gas_viz.py (hunk 2 above)
Comment (owocki) [reply app/gas/management/commands/output_gas_viz.py]: updated

**Comment by owocki** [comment]
updated per code review

**Pull request merged**